maths/pi_generator.py

```diff
@@ -0,0 +1,80 @@
+def calculate_pi(limit: int) -> str:
+    """
+    https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80
+    Leibniz Formula for Pi
+
+    The Leibniz formula is the special case arctan 1 = 1/4 Pi .
+    Leibniz's formula converges extremely slowly: it exhibits sublinear convergence.
+
+    Convergence (https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80#Convergence)
+
+    We cannot try to prove against an interrupted, uncompleted generation.
+    https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80#Unusual_behaviour
+    The errors can in fact be predicted;
+    but those calculations also approach infinity for accuracy.
+    For simplicity' sake, let's just compare against known values.
+
+    >>> calculate_pi(15)
+    '3.141592653589793'
+
+    To further proof, since we cannot predict errors or interrupt an infinite generation
+    or interrupt any alternating series, here some more tests.
+
+    >>> calculate_pi(50)
+    '3.14159265358979323846264338327950288419716939937510'
+
+    >>> calculate_pi(80)
+    '3.14159265358979323846264338327950288419716939937510582097494459230781640628620899'
+    """
```

Comment:
for viewer understanding, you can add one more line -

''In the Leibniz formula for calculating pi, the variables q, r, t, k, n, and l are used for the iteration process''